### PR TITLE
Fix deprecated ReflectiveClassBuildItem construct method and replace with builder()

### PR DESCRIPTION
**Pull request by zhfeng**
related to https://github.com/quarkusio/quarkus/pull/31222 and ReflectiveClassBuildItem construct methods have been marked for removal.

### CODE REVIEW

File: extensions/jdbc/jdbc-mysql/deployment/src/main/java/io/quarkus/jdbc/mysql/deployment/MySQLJDBCReflections.java [hunk 1]
```diff
@@ -54,74 +54,115 @@ public final class MySQLJDBCReflections {
...
+                ReflectiveClassBuildItem.builder(SSLParamsException.class.getName()).methods(false).fields(false).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(StatementIsClosedException.class.getName()).methods(false)
+                .fields(false).build());
+        reflectiveClass.produce(
+                ReflectiveClassBuildItem.builder(StreamingNotifiable.class.getName()).methods(false).fields(false).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder(UnableToConnectException.class.getName()).methods(false)
+                .fields(false).build());
         reflectiveClass
-                .produce(new ReflectiveClassBuildItem(false, false, UnsupportedConnectionStringException.class.getName()));
-        reflectiveClass.produce(new ReflectiveClassBuildItem(false, false, WrongArgumentException.class.getName()));
```
Comment (zakkak) [inline extensions/jdbc/jdbc-mysql/deployment/src/main/java/io/quarkus/jdbc/mysql/deployment/MySQLJDBCReflections.java:121]: These could be combined in a single builder

File: extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleMetadataOverrides.java [hunk 2]
```diff
@@ -47,23 +47,39 @@ public final class OracleMetadataOverrides {
...
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder("oracle.net.ano.AuthenticationService").constructors(true)
+                .methods(false).fields(false).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder("oracle.net.ano.DataIntegrityService").constructors(true)
+                .methods(false).fields(false).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder("oracle.net.ano.EncryptionService").constructors(true)
+                .methods(false).fields(false).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder("oracle.net.ano.SupervisorService").constructors(true)
+                .methods(false).fields(false).build());
+        reflectiveClass.produce(ReflectiveClassBuildItem.builder("oracle.jdbc.driver.Message11").constructors(true)
+                .methods(false).fields(false).build());
```
Comment (zakkak) [inline extensions/jdbc/jdbc-oracle/deployment/src/main/java/io/quarkus/jdbc/oracle/deployment/OracleMetadataOverrides.java:69]: These can be combined in as single builder

**Review by zakkak — APPROVED** [review]
LGTM in general.

To further improve the PR I suggest:

1. removing the `methods(false)` and `fields(false)` invocations though as `methods` and `fields` are `false by default
2. replacing `methods(true)` and `fields(true)` with `methods()` and `fields()`
3. using builder(Class<?>... classes) instead of builder(String... classNames) to get rid of all the `getName()` invocations
4. combining ReflectiveClassBuildItems with the same configuration in a single builder

**Comment by zhfeng** [comment]
Thanks @zakkak for reviewing. I just used [Structural Replace](https://www.jetbrains.com/help/idea/structural-search-and-replace.html) in IDEA to do these refactors automatically.

**Comment by zhfeng** [comment]
`constructors(true)` should also be removed since the `constructors` is default by `true`?

**Comment by gsmet** [comment]
My opinion is that it's a good first step.

You could probably do:
- removing the methods(false) and fields(false) invocations though as methods and fields are `false by default
- replacing methods(true) and fields(true) with methods() and fields()
easily.

For the rest, maybe keep it the way it is, and we can improve when specifically adjusting an extension.

Also we need to be careful: when we pass an actual string to the builder (`"my.Class.class"`), it's actually because we want to pass a string (either because we don't want to load the class or because the class is not visible).

**Comment by geoand** [comment]
> For the rest, maybe keep it the way it is, and we can improve when specifically adjusting an extension.

+100

**Comment by zhfeng** [comment]
@zakkak @gsmet 
> constructors(true) should also be removed since the constructors is default by true?

Is it right?

**Comment by zakkak** [comment]
> @zakkak @gsmet
> 
> > constructors(true) should also be removed since the constructors is default by true?
> 
> Is it right?

Yes, I think that could be removed as well.

**Comment by zhfeng** [comment]
>- removing the methods(false) and fields(false) invocations though as methods and fields are `false` by default
>- replacing methods(true) and fields(true) with methods() and fields()
>- constructors(true) should also be removed since the constructors is default by true

Done

**Review by zakkak — APPROVED** [review]
Still looks good to me!

Thanks for the additional changes.

**Comment by gsmet** [comment]
Thanks a lot for taking care of this. I had it in the back of my head and happy someone else tackled it :).

**Pull request merged**